source/slang/slang-ir-reachability.cpp

```diff
@@ -54,6 +55,45 @@ namespace Slang
         }
 
     }
+    
+    bool ReachabilityContext::isInstReachable(IRInst* inst1, IRInst* inst2)
+    {
+        // If inst1 and inst2 are in the same block,
+        // we test if inst2 appears after inst1.
+        if (inst1->getParent() == inst2->getParent())
+        {
+            for (auto inst = inst1->getNextInst(); inst; inst = inst->getNextInst())
+            {
+                if (inst == inst2)
+                    return true;
+            }
+        }
+
+        // Special cases
+
+        // Target switches; treat as reachable from its cases
+        if (auto tswitch = as<IRTargetSwitch>(inst2))
+        {
+            HashSet<IRInst*> blocks;
+            for (Slang::UInt i = 0; i < tswitch->getCaseCount(); i++)
+                blocks.add(tswitch->getCaseBlock(i));
+
+            IRInst* tmp = inst1->getParent();
```

Comment:
You may want to call getBlock instead of getParent here. GetBlock returns the enclosing block of the instruction. For instructions in spirv asm blocks, their parent is the spirv inst instead of the enclosing block.

Reply:
Where is the `getBlock()` method? It doesn't look like `IRInst` has that.